### ISSUE TITLE
TypeError: ufunc 'isnan' not supported for the input types

### ISSUE THREAD

**Issue by jazberna1** [koszullab/chromosight#41]
Hello Cyril,

I have run chromosight 1.4.1 and I encountered the following error. This is the full log. I have used the same input .cool file on 1.4.0. Could you possibly think on something I'm doing wrong? 

_pearson set to 0.3 based on config file.
min_separation set to 5000 based on config file.
max_perc_undetected set to 50.0 based on config file.
max_perc_zero set to 10.0 based on config file.
WARNING: Detection on interchromosomal matrices is expensive in RAM
Matrix already balanced, reusing weights
Preprocessing sub-matrices...
 [====================] 100.0% EBV-EBV
Detecting patterns...
 [--------------------] 0.0% Kernel: 0, Iteration: 0
 [====================] 100.0% Kernel: 0, Iteration: 0
Traceback (most recent call last):
  File "/mnt/lustre/scratch/SOFTWARE/miniconda3/bin/chromosight", line 8, in <module>
    sys.exit(main())
  File "/mnt/lustre/scratch/SOFTWARE/miniconda3/lib/python3.8/site-packages/chromosight/cli/chromosight.py", line 950, in main
    cmd_detect(args)
  File "/mnt/lustre/scratch/SOFTWARE/miniconda3/lib/python3.8/site-packages/chromosight/cli/chromosight.py", line 793, in cmd_detect
    pval_mask = np.isnan(all_coords.pvalue)
  File "/mnt/lustre/scratch/SOFTWARE/miniconda3/lib/python3.8/site-packages/pandas/core/generic.py", line 1935, in __array_ufunc__
    return arraylike.array_ufunc(self, ufunc, method, *inputs, **kwargs)
  File "/mnt/lustre/scratch/SOFTWARE/miniconda3/lib/python3.8/site-packages/pandas/core/arraylike.py", line 358, in array_ufunc
    result = getattr(ufunc, method)(*inputs, **kwargs)
TypeError: ufunc 'isnan' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''_

This is how I have run chromosight

export PATH=/mnt/lustre/scratch/SOFTWARE/miniconda3/bin:$PATH

```
/mnt/lustre/scratch/SOFTWARE/miniconda3/bin/chromosight \
detect \
--threads=1 \
--inter \
--min-dist 50000 --max-dist 200000 \
/mnt/lustre/scratch/test.matrix.cool \
loops_output

```

Thanks so much
Jorge

**Comment by cmdoret** [comment]
Hi Jorge,

Unfortunately I am unable to replicate your problem when running the same command on two separate datasets.
Note that using --min-dist and --max-dist will restrict the loop size, but this is only meaning full for intra-chromosomal matrices. Loop pattern will still be scanned on the entire inter-chromosomal matrices.

What is your version of pandas ? It works with v1.1.3, but I could issue a fix to make it compatible with other versions, if that's the problem.

Cyril

**Comment by jazberna1** [comment]
Hello Cyril,

I have matched my pandas to be 1.1.3 as yours and rerun chromosight 1.4.1. I'll let you know what happens.

Many thanks
Jorge

**Comment by jazberna1** [comment]
Hello Cyril,

I rerun with pandas  1.1.3 but still got the same error. I'll try with chromosight 1.4.0 instead.

Thanks
Jorge

**Comment by cmdoret** [comment]
OK then, since I cannot reproduce the error, I'll close it for now. Don't hesitate to reopen if you encounter this error on a different dataset. If so, it would be nice to see your python environment to find out what package is causing the issue.

Best,
Cyril

**Comment by cmacphillamy** [comment]
Hello Cyril,

I have gotten the same error. I ran chromosight on a matrix enhanced with DeepHiC and the original. The matrix that was enhanced did work, and the original failed. Is it likely that the original matrix is just too sparse to identify any loops?

Kind regards,
Callum

**Comment by cmdoret** [comment]
Hi @cmacphillamy,

Do you get the exact same error ?
I have tried running `chromosight detect` on an empty matrix, but this just returns "No patterns detected" as expected, so I do not think it has to do with sparsity.

I don't really know what the issue could be, but if you were comfortable with sharing the original cool file, I could investigate a bit.

Best,
Cyril

**Comment by GeorgyChistov** [comment]
Hello! @cmdoret 

I got the same problem. It disappears when I use --pearson==0.1, but appears when I set this parameter from 0.2 to 0.9. 
If --pearson is 1 or higher, the chromosight said "No patterns detected", as you said.


UPD:
I revealed that pearson between 0.1 and 0.2 works good. But from 0.2 and 0.9 it raises error

UPD2:
The change of resolution allowed to increase the pearson parameter, but I still get the error when it is equal to 0.5, for example.

**Comment by cmdoret** [comment]
Hi @GeorgyChistov, since I cannot reproduce the issue on my end, would it be possible for you to share the dataset on which it happens ? It is hard to figure out the cause if I can't reproduce the error.

**Comment by GeorgyChistov** [comment]
@cmdoret  of course, here is command

chromosight detect   30C_15min.mcool::/resolutions/3200 CIDs_30C_15min_res3200 --pattern borders --win-fmt=npy [--pearson=0.199

I am sorry, I cannot attach the data. Here is link to it

https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM4585137 

Additionally, I have another question to you, as to the developer. How chromosight  specify the size of its scanning window? For example, we have borders on the diagonal with size 3x3 bins and borders with size 20x20. Will the chromosight find both them? So does it somehow enumerate the size of scanning window?

**Comment by cmdoret** [comment]
Hi @GeorgyChistov I downloaded the data and ran the command, but could not reproduce your error.
It may have something to do with a particular version of numpy / pandas on your machine.

Since Chromosight is distributed on bioconda, [a docker container](https://quay.io/repository/biocontainers/chromosight?tab=info) is built automatically by biocontainers for each of its release. To isolate it from your system libraries, you can run chromosight from the container if you have docker installed:

```sh
docker run -v "$PWD:/data" quay.io/biocontainers/chromosight:1.6.1--pyhdfd78af_0 chromosight detect \
  /data/GSM4585137_30C_15min.mcool::/resolutions/3200 \
  /data/CIDs_30C_15min_res3200 \
  --pattern borders \
  --win-fmt=npy \
  --pearson=0.199
```

As for your question: The windows size is fixed to a default value (you can change it with `--win-size`) which we found worked pretty well for most of our use cases. Generally speaking, Chromosight should find borders larger than the windows without problem, but may miss borders smaller than the window. If you want to detect very small borders, reducing it will probably increase sensitivity, however this will also cause many false positives.

**Comment by cmdoret** [comment]
I figured out the problem @jazberna1 @GeorgyChistov this was indeed due to a newer version of pandas.
Briefly, using `np.isnan` on a pandas series with dtype object now returns an error, wheres it worked before.

I fixed it in 063ba467c8584abbb1e2ce3a2def95c9e23b8486 and released 1.6.2. Now Chromosight works with newer versions of pandas and is compatible with python 3.10.
